var/spack/repos/builtin/packages/upcxx/package.py

```diff
@@ -64,26 +49,27 @@ def setup_environment(self, spack_env, run_env):
         if 'platform=cray' in self.spec:
             spack_env.set('GASNET_CONFIGURE_ARGS', '--enable-mpi=probe')
             if "scheduler=slurm" in self.spec:
-                tty.warn("CROSS has been automatically set to %s."
-                         % 'cray-aries-slurm')
                 spack_env.set('CROSS', 'cray-aries-slurm')
             elif "scheduler=alps" in self.spec:
-                tty.warn("CROSS has been automatically set to %s."
-                         % 'cray-aries-alps')
                 spack_env.set('CROSS', 'cray-aries-alps')
         if '+cuda' in self.spec:
             spack_env.set('UPCXX_CUDA', '1')
             spack_env.set('UPCXX_CUDA_NVCC', self.spec['cuda'].prefix.bin.nvcc)
 
+        run_env.set('UPCXX_INSTALL', self.prefix)
+        run_env.set('UPCXX', self.prefix.bin.upcxx)
+        if 'platform=cray' in self.spec:
+            run_env.set('UPCXX_GASNET_CONDUIT', 'aries')
```

Comment:
Please additionally set `run_env.set('UPCXX_NETWORK', 'aries')`, as the former spelling is deprecated starting in v2019.9.0 and may be removed in a future version.